### ISSUE TITLE
JS stops working after the first transition to another page

### ISSUE THREAD

**Issue by sergeyfilimonov** [gridsome/gridsome#289]
#288 # Description

When built, one transition to another page ruins all JavaScript — I can't go back or switch to another page, or animations stop working. This happens only on desktop.

In console, you can see:

```
Uncaught (in promise) DOMException: Failed to execute 'appendChild' on 'Node': This node type does not support this method.
    at Object.appendChild (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:40392)
    at p (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:52966)
    at f (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:52703)
    at h (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:53049)
    at E (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:56391)
    at E (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:56326)
    at E (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:56326)
    at E (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:56326)
    at E (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:56326)
    at a.__patch__ (https://sergeyfilimonovcom-jb23svs5l.now.sh/assets/js/app.5400ccbf.js:7:56749)
app.5400ccbf.js:7 TypeError: Cannot read property '_isDestroyed' of undefined
    at destroy (app.5400ccbf.js:7)
    at w (app.5400ccbf.js:7)
    at w (app.5400ccbf.js:7)
    at a.__patch__ (app.5400ccbf.js:7)
    at a.t.$destroy (app.5400ccbf.js:7)
    at destroy (app.5400ccbf.js:7)
    at w (app.5400ccbf.js:7)
    at a.__patch__ (app.5400ccbf.js:7)
    at a.t.$destroy (app.5400ccbf.js:7)
    at destroy (app.5400ccbf.js:7)
```

### Steps to reproduce

Go to https://sergeyfilimonovcom-jb23svs5l.now.sh and surf around the website. You are going to notice that routing will not be working after the first transition.

If you delete `<Messenger/>` from https://github.com/sergeyfilimonov/sergeyfilimonov.com/blob/master/src/layouts/Default.vue the problem goes away but I can't get WHY. Sometimes it is working, sometimes not. For instance, if you just move the component right after `<div class"layout">` the bug is gone, here is the proof: https://sergeyfilimonovcom-bd6stk8jn.now.sh. If you move it back, right before `</div>` it stops working.

You can see the repo here https://github.com/sergeyfilimonov/sergeyfilimonov.com

### Environment

<pre><code>
Libs:
- gridsome version: 5.0.4

Browser:
- [x] Chrome (desktop) version XX
- [x] Safari (desktop) version XX
 
For Tooling issues:
- Node version: 8.12.0 
- Platform:  Mac

**Comment by cossssmin** [comment]
Not sure if this is causing it (on my phone), but it should be `!==` :

https://github.com/sergeyfilimonov/sergeyfilimonov.com/blob/d73e301c00fdd6b5bd9ebd561889cfabd7c6b6be/src/components/Messenger.vue#L11

**Comment by hjvedvik** [comment]
Seems like Vue fails to hydrate. Try running webpack in dev mode to get a better error message. See the [troubleshooting](https://gridsome.org/docs/troubleshooting#debugging-hydration-errors) page for an example on how to do that.

**Comment by robaxelsen** [comment]
@sergeyfilimonov have you had a chance to troubleshoot further, re @hjvedvik comment above? :point_up:

**Comment by sergeyfilimonov** [comment]
@robaxelsen not yet, https://gridsome.org/docs/troubleshooting#debugging-hydration-errors is not working for me from Russia :/ The Gridsome site is located on some servers which are blocked by our government because of Telegram. Will try to check that solution when the site opens for me.

**Comment by Al-Rozhkov** [comment]
@sergeyfilimonov try Tor browser (https://www.torproject.org/). I can visit gridsome.org that way.

**Comment by robaxelsen** [comment]
@sergeyfilimonov no worries, and sorry to hear about the sensorship issue. Here is an archived version (mnessed up formatting though, but content available at bottom): http://archive.is/b1ado

And if that doesn't work, here is a screenshot of the page:
![Screenshot from 2019-04-02 10-58-15](https://user-images.githubusercontent.com/13132899/55389929-7a4d6c00-5536-11e9-95e8-7e5c1222d2c3.png)

**Comment by hjvedvik** [comment]
@sergeyfilimonov Sometimes I copy the `<body>` element in the DOM tree as an HTML string (not the markup). Once after Vue has rendered the page and once after a refresh with JS disabled. And then run the HTML through [diffchecker](https://www.diffchecker.com/) to find extra elements in the HTML from the non-JS DOM tree. Differences in attributes should be fine, it's the extra elements  which causes trouble.

**Comment by sergeyfilimonov** [comment]
Sorry but where should I add this?
```js
chainWebpack (config) {
  config.mode('development')
}
#
```

**Comment by hjvedvik** [comment]
@sergeyfilimonov It should be in your `gridsome.config.js` file.

**Comment by sergeyfilimonov** [comment]
@hjvedvik I wrote above:

> For instance, if you just move the component right after <div class"layout"> the bug is gone, here is the proof: https://sergeyfilimonovcom-bd6stk8jn.now.sh. If you move it back, right before </div> it stops working.

Now I placed it back to `</div>` and have no problems with routing and other stuff! It seems adding:
```js
chainWebpack (config) {
    config.mode('development')
  }
```
really helped but I don't know why. Thanks!

**Comment by robaxelsen** [comment]
Closing, as it seems OPs original question has been answered. Feel free to reopen if you have more questions, or open a new issue (preferable).

**Comment by sergeyfilimonov** [comment]
@hjvedvik I was too soon to speak and got a very similar error but this time with no errors in the console https://github.com/gridsome/gridsome/issues/364

**Comment by ricardov03** [comment]
> @hjvedvik I wrote above:
> 
> > For instance, if you just move the [Messenger] component right after `<div class"layout">` the bug is gone, here is the proof: https://sergeyfilimonovcom-bd6stk8jn.now.sh. If you move it back, right before `</div>` it stops working.
> 
> Now I placed it back to `</div>` and have no problems with routing and other stuff! It seems adding:
> 
> ```js
> chainWebpack (config) {
>     config.mode('development')
>   }
> ```
> 
> in the `module.exports` of `gridsome.config.js` really helped but I don't know why. Thanks!

Sorry, Can anybody help me to figure out why config.mode('development') it's no working in the build for me?
